parts/22-query.sh

```diff
@@ -4905,3 +4905,77 @@ query_tpt-tool-memory() { ##? [--startyear=<YYYY>] [--endyear=<YYYY>] [--formula
 	EOF
 }
 
+query_tools-usage-per-month() { ##? [--startmonth=<YYYY>-<MM>] [--endmonth=<YYYY>-<MM>] [--tools=<tool1,tool2,...>] [--short_tool_id] [--super_short_tool_id] [--no_version]: By default, startmonth is 1 year ago and end month is current month. tool1, tool2 etc. should correspond to the tool_id with the same format as requested: toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0,Cut1 for default, devteam/bowtie2/bowtie2/2.5.0+galaxy0,Cut1 for --short_tool_id, bowtie2/2.5.0+galaxy0,Cut1 for --super_short_tool_id etc...
+	meta <<-EOF
+		AUTHORS: lldelisle
+		ADDED: 22
+	EOF
+	handle_help "$@" <<-EOF
+		Tools Usage Tracking: cpu-hours and nb_users by Month-Year.
+
+		    $ gxadmin query tools-usage-per-month --super_short_tool_id --no_version --tools bowtie2,Cut1 --startmonth=2023-03 --endmonth 2023-08
+			   month    | cpu_hours | tool_id | nb_users 
+			------------+-----------+---------+----------
+			 2023-08-01 |    326.88 | bowtie2 |        1
+			 2023-08-01 |    469.27 | bowtie2 |        1
+			 2023-07-01 |      0.01 | Cut1    |        2
+			 2023-07-01 |     20.04 | bowtie2 |        1
+			 2023-06-01 |      0.04 | Cut1    |        2
+			 2023-06-01 |    271.16 | bowtie2 |        3
+			 2023-05-01 |    732.74 | bowtie2 |        3
+			 2023-04-01 |      1.55 | Cut1    |        2
+			 2023-04-01 |    426.32 | bowtie2 |        2
+			 2023-03-01 |      0.00 | Cut1    |        1
+			 2023-03-01 |    437.31 | bowtie2 |        1
+			 2023-03-01 |    506.71 | bowtie2 |        2
+			(12 rows)
+	EOF
+	
+	filter_by_time_period=""
+	if [[ -n $arg_startmonth ]]; then
+		filter_by_time_period="date_trunc('month', job.create_time AT TIME ZONE 'UTC') >= '$arg_startmonth-01'::date"
+	else
+		filter_by_time_period="job.create_time > NOW() - interval '1 year'"
+	fi
+	if [[ -n $arg_endmonth ]]; then
+		filter_by_time_period="$filter_by_time_period AND date_trunc('month', job.create_time AT TIME ZONE 'UTC') <= '$arg_endmonth-01'::date"
+	fi
+	tool_id="job.tool_id"
+	if [[ -n $arg_short_tool_id ]]; then
+		tool_id="regexp_replace(job.tool_id, '.*toolshed.*/repos/', '')"
+	fi
+	if [[ -n $arg_super_short_tool_id ]]; then
+		tool_id="regexp_replace(job.tool_id, '.*toolshed.*/repos/[^/]*/[^/]*/', '')"
+	fi
+
+	if [[ -n $arg_no_version ]]; then
+		tool_id="regexp_replace(${tool_id}::TEXT, '/[0-9.a-z+-]+$', '')"
+	fi
+	filter_tool=""
+	if [[ -n $arg_tools ]]; then
```

Comment:
```suggestion
	if [[ $arg_tools != "<tool1,tool2,...>" ]]; then
```

you could do something like this instead?